components/feature/intent/src/main/java/mozilla/components/feature/intent/IntentProcessor.kt

```diff
@@ -28,14 +28,16 @@ typealias IntentHandler = (Intent) -> Boolean
  * @property useDefaultHandlers Whether or not the built-in handlers should be used.
  * @property openNewTab Whether a processed intent should open a new tab or
  * open URLs in the currently selected tab.
+ * @property isPrivate Whether a processed intent should open a new tab as private
  */
 class IntentProcessor(
     private val sessionUseCases: SessionUseCases,
     private val sessionManager: SessionManager,
     private val searchUseCases: SearchUseCases,
     private val context: Context,
     private val useDefaultHandlers: Boolean = true,
-    private val openNewTab: Boolean = true
+    private val openNewTab: Boolean = true,
+    var isPrivate: Boolean = false
```

Comment:
Apps usually have one IntentProcessor to handle all incoming intents, so if this becomes a contructor argument, you would need an IntentProcessor instantiated for private and non-private sessions.

This might be a case where we want to allow the app to delegate whether to make this a private session:
```kotlin
class IntentProcessor(
  /* other deps */
  private val privateSession: () -> Boolean = { false }
) {}
```

This would allow the app to decide (maybe based on a pref?) when a session should be private/regular. At the calling site you'd have this:

```kotlin
val session = createSession(url, private = privateSession.invoke(), source = Source.ACTION_VIEW)
```

I think we can do better than this as well, but I'm not quite sure what.. 🤔

Reply:
This is currently the case but actually there's no reason for the `IntentProcessor` to be a global singleton. It could just live in the scope of the activity that handles the incoming Intents?